core/lib/multistore/storeCodeFromRoute.ts

```diff
@@ -0,0 +1,48 @@
+import { RawLocation } from 'vue-router'
+import config from 'config'
+import { LocalizedRoute } from './../types'
+import { getNormalizedPath, getUrl } from './helpers'
+import { getExtendedStoreviewConfig } from '.'
+import cloneDeep from 'lodash-es/cloneDeep'
+
+const isMatchingByPath = (matchedRouteOrUrl, store) => {
+  const normalizedPath = getNormalizedPath(matchedRouteOrUrl)
+  return normalizedPath.startsWith(`${store.url}/`) || normalizedPath === store.url
+}
+
+const isMatchingByDomainAndPath = (matchedRouteOrUrl, store) => {
+  const url = getUrl(matchedRouteOrUrl)
+  return url.startsWith(`${store.url}/`) || url === store.url
+}
+
+const isMatchingWithAppendStoreCode = (matchedRouteOrUrl, store) => {
+  const clonedStoreView = cloneDeep(store)
+  clonedStoreView.url = `/${store.storeCode}`
+  return isMatchingByPath(matchedRouteOrUrl, clonedStoreView) || isMatchingByDomainAndPath(matchedRouteOrUrl, clonedStoreView)
```

Comment:
I am wondering only by names of these functions.
- Right part cannot be `true` if left part is false
- Right part cannot be used if left part is true

So could we get rid of it?